### PR TITLE
Support for all arguments in make_zim_file()

### LINKED ISSUE #20: More flexible make_zim_file() in zim.py
The current function doesn't support the extra arguments that the to_zimwriterfs_args() method takes in. It would be better if we pass kwargs to to_zimwriterfs_args() from make_zim_file()

### PR DESCRIPTION
**Pull request by satyamtg**
Fixes #20 by making make_zim_file() take in kwargs which are passed to to_zimwriterfs_args(). Also, made it take the zimwriterfs binary location from the object of ZimInfo that's passed as it would allow to change it easily if required. Also added some docstrings regarding the arguments.

This one changes the API

### CODE REVIEW

File: src/zimscraperlib/zim.py [hunk 1]
```diff
@@ -100,11 +100,26 @@ def to_zimwriterfs_args(
...
+def make_zim_file(build_dir, output_dir, zim_fname, zim_info, **kwargs):
+    """ runs zimwriterfs 
+    
+    The function takes in the following arguments
+        build_dir: The directory to convert into a ZIM file
+        output_dir: The directory where the ZIM file would be saved
+        zim_fname: The name of the ZIM file
+        zim_info: An object of class ZimInfo
+    
+    Also supports the following keyword arguments that are passed to to_zimwriterfs_args()
```
Comment (rgaudin) [inline src/zimscraperlib/zim.py:112]: Good to mention is supports `to_zimwriterfs_args()` but that's enough IMO. That's not the place to describe those. We could do that in the other method but it's advanced feautres anyway.

File: src/zimscraperlib/zim.py [hunk 2]
```diff
@@ -100,11 +100,26 @@ def to_zimwriterfs_args(
         return arg_list
 
 
-def make_zim_file(build_dir, output_dir, zim_fname, zim_info):
-    """ runs zimwriterfs """
+def make_zim_file(build_dir, output_dir, zim_fname, zim_info, **kwargs):
+    """ runs zimwriterfs
```
Comment (rgaudin) [inline src/zimscraperlib/zim.py:104]:
``` py
""" Creates a zim file using zimwriterfs

    Arguments:
        build_dir: The directory to convert into a ZIM file
        output_dir: The directory where the ZIM file would be saved
        zim_fname: The name of the ZIM file
        zim_info: An object of class ZimInfo

        **kwargs: passed directly to to_zimwriterfs_args() """
```

File: src/zimscraperlib/zim.py [hunk 3]
```diff
@@ -100,11 +100,26 @@ def to_zimwriterfs_args(
...
+        inflateHtml: Whether try to inflate HTML files
+        uniqueNamespace: Put everything in namespace 'A'
+        withoutFTIndex: No fulltext index
+        minChunkSize: Number of bytes per ZIM cluster
+        redirects: Path to file containing redirects
+    """
     args = (
-        [ZimInfo.zimwriterfs_path]
-        + zim_info.to_zimwriterfs_args()
+        [zim_info.zimwriterfs_path]
```
Comment (rgaudin) [inline src/zimscraperlib/zim.py:121]: We could use this PR to remove this `zimwriterfs_path` non-sense and just replace it with `["/usr/bin/env", "zimwriterfs"]`

File: src/zimscraperlib/zim.py (hunk 2 above)
Comment (satyamtg) [reply src/zimscraperlib/zim.py:104]: Okay

**Pull request merged**